[SPARK-38800][DOCS][PYTHON] Explicitly document the supported pandas version. (#36095)
apache/spark: python/docs/source/reference/index.rst

This page lists an overview of all public PySpark modules, classes, functions and methods.

> Note that pandas has different behavior per its version, and pandas API on Spark tries to match the behavior of the latest pandas release.

Reviewer: ```suggestion
pandas API on Spark follows the API specifications of latest pandas release.
```
